src/value-parser.js

```diff
@@ -276,27 +276,40 @@ function valueParse(parser, metaData, options, callback) {
             return parser.readUInt8((sign) => {
               sign = sign === 1 ? 1 : -1;
 
-              let readValue;
-              switch (dataLength - 1) {
-                case 4:
-                  readValue = parser.readUInt32LE;
-                  break;
-                case 8:
-                  readValue = parser.readUNumeric64LE;
-                  break;
-                case 12:
-                  readValue = parser.readUNumeric96LE;
-                  break;
-                case 16:
-                  readValue = parser.readUNumeric128LE;
-                  break;
-                default:
-                  return parser.emit('error', new Error(sprintf('Unsupported numeric size %d', dataLength - 1)));
+              if (options.returnDecimalAndNumericAsString) {
+                return parser.readBuffer(dataLength - 1, (buffer) => {
+                  let value = convertLEBytesToString(buffer);
+                  if (metaData.scale > 0) {
+                    value = value.padStart(metaData.scale + 1, '0');
```

Comment:
`padStart` will work only if the value is zero, as `convertLEBytesToString` gives padded value for everything else. Can you add a check for calling `padStart`?